src/python/pants/backend/jvm/targets/scala_library.py

```diff
@@ -82,3 +89,37 @@ def java_sources(self):
         raise TargetDefinitionException(self, 'No such java target: {}'.format(spec))
       targets.append(target)
     return targets
+
+  @property
+  def scalac_plugins(self):
+    """The names of compiler plugins to use when compiling this target with scalac.
+    :return: See constructor.
+    :rtype: list of strings.
+    """
+    if self._scoverage:
```

Comment:
Rather than fully overriding the set of plugins, this should merge both lists. Ditto the plugin args and options.

Reply:
the function `get_scalac_plugins` in scala_coverage_platform.py does that. It merges the list if non empty (otherwise creates a new one)

Reply:
Ok. That's a bit confusing, because if someone were to change line 102 here, they would need to remember to change the subsystem code as well. IMO, the merging code should live in here to avoid that issue.